test/udp_sender.js

```diff
@@ -249,4 +250,21 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', done => {
+        sender._host = 'foo.bar.com';
+        sender._port = 1234;
+        new Tracer(
+            'test-service-name',
+            new RemoteReporter(sender),
+            new ConstSampler(true)
+        ).startSpan('testSpan').finish();
+        let oldLog = console.log;
+        sender.flush();
+        console.log = message =>  {
```

Comment:
No I don't believe so. In the context of replacing `console.log`, the code in question is:

```js
this._client.on('error', err => {
    console.log(`error sending span: ${err}`)
})
```

Now, there is now way this code will be called before `socket.send` has a chance to execute. Now, according to the Node dgram docs where it says, *Note that DNS lookups delay the time to send for at least one tick of the Node.js event loop.* During that time, the next line of code, the code that replaces the logger to our mock one, will have a chance to execute.